pkg/eventing/deployment_test.go

```diff
@@ -0,0 +1,312 @@
+		name      string
+		givenEnvs map[string]string
+		wantEnvs  map[string]string
+	}{
+		{
+			name: "REQUEST_TIMEOUT is not set, the default value should be taken",
+			givenEnvs: map[string]string{
+				"PUBLISHER_REQUESTS_CPU": "64m",
+			},
+			wantEnvs: map[string]string{
+				"REQUEST_TIMEOUT": "5s", // default value
+			},
+		},
+		{
+			name: "REQUEST_TIMEOUT should be set",
+			givenEnvs: map[string]string{
+				"PUBLISHER_REQUEST_TIMEOUT": "10s",
+			},
+			wantEnvs: map[string]string{
+				"REQUEST_TIMEOUT": "10s",
+			},
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			for k, v := range tc.givenEnvs {
+				t.Setenv(k, v)
+			}
+			backendConfig := env.GetBackendConfig()
+			envVars := getEventMeshEnvVars("test-name", backendConfig.PublisherConfig)
+
+			// ensure the right envs were set
+			for index, val := range tc.wantEnvs {
+				gotEnv := findEnvVar(envVars, index)
+				assert.NotNil(t, gotEnv)
+				assert.Equal(t, val, gotEnv.Value)
+			}
+		})
+	}
+}
+
+// natsBackendAssertions checks that the NATS-specific data was set in the NewNATSPublisherDeployment.
+func natsBackendAssertions(t *testing.T, publisherName string, deployment appsv1.Deployment) {
+	container := findPublisherContainer(publisherName, deployment)
+	assert.NotNil(t, container)
+
+	streamName := findEnvVar(container.Env, "JS_STREAM_NAME")
+	assert.Equal(t, streamName.Value, "kyma")
+	url := findEnvVar(container.Env, "NATS_URL")
+	assert.Equal(t, url.Value, natsURL)
+
+	// check the affinity was set
+	affinityLabels := deployment.Spec.Template.Spec.Affinity.PodAntiAffinity.PreferredDuringSchedulingIgnoredDuringExecution[0].PodAffinityTerm.LabelSelector.MatchLabels
+	for _, val := range affinityLabels {
+		assert.Equal(t, val, publisherName)
+	}
+}
+
+// eventMeshBackendAssertions checks that the eventmesh-specific data was set in the NewEventMeshPublisherDeployment.
+func eventMeshBackendAssertions(t *testing.T, publisherName string, deployment appsv1.Deployment) {
+	container := findPublisherContainer(publisherName, deployment)
+	assert.NotNil(t, container)
+
+	// check eventmesh-specific env variables
+	eventMeshNamespace := findEnvVar(container.Env, "BEB_NAMESPACE")
+	assert.Equal(t, eventMeshNamespace.Value, fmt.Sprintf("%s$(BEB_NAMESPACE_VALUE)", eventMeshNamespacePrefix))
+
+	// check the affinity is empty
+	assert.Empty(t, deployment.Spec.Template.Spec.Affinity)
+}
+
+// findPublisherContainer gets the publisher proxy container by its name.
+func findPublisherContainer(publisherName string, deployment appsv1.Deployment) v1.Container {
+	var container v1.Container
+	for _, c := range deployment.Spec.Template.Spec.Containers {
+		if strings.EqualFold(c.Name, publisherName) {
+			container = c
+		}
+	}
+	return container
+}
+
+// findEnvVar returns the env variable which has `name == envVar.Name`,
+// or `nil` if there is no such env variable.
+func findEnvVar(envVars []v1.EnvVar, name string) *v1.EnvVar {
```

Comment:
(nit) We can move it to test utils pkg so later other tests can use it as well?

Reply:
I am already moving it for eventTypePrefix implementation, hence it'll be in another PR.